test/src/main/java/org/springframework/security/test/web/support/WebTestUtils.java

```diff
@@ -114,6 +115,21 @@ public static void setCsrfTokenRepository(HttpServletRequest request,
 		}
 	}
 
+	/**
+	 * Sets the {@link RequestMatcher} for the specified {@link HttpServletRequest}.
+	 *
+	 * @param request the {@link RequestMatcher} to obtain the
+	 * {@link RequestMatcher}
+	 * @param requestMatcher the {@link RequestMatcher} to set
+	 */
+	public static void setCsrfRequestMatcher(HttpServletRequest request,
```

Comment:
We cannot do this because it dirties the context since additional requests without a jwt token would also ignore CSRF protection. The CSRF request should be disabled for just this request.

It might be better to:

1) Update CsrfFilter.shouldNotFilter to return true if a specific request attribute is set
2) Modify the code to set the attribute when jwt is being invoked